### PR TITLE
vertical housing cooperative base

### PR DESCRIPTION
**Pull request by robinkeunen**

### CODE REVIEW

File: housing_cooperative_base/views/cluster.xml [hunk 1]
```diff
@@ -19,12 +19,12 @@
                     <notebook>
                         <page name="housings" string="Housings">
                             <group>
-                                <field name="housing_ids" nolabel="1"/>
+                                <field name="housing_ids" nolabel="1" widget="many2many"/>
```
Comment (robinkeunen) [inline housing_cooperative_base/views/cluster.xml]: C'est pas le widget par défaut?

File: housing_cooperative_base/views/building.xml [hunk 2]
```diff
@@ -10,6 +10,13 @@
         <field name="arch" type="xml">
             <form string="builging_form">
                 <sheet>
+                    <div class="oe_button_box" name="button_box">
+                        <button name="toggle_active" type="object"
+                            class="oe_stat_button" icon="fa-archive">
+                            <field name="active" widget="boolean_button"
```
Comment (robinkeunen) [inline housing_cooperative_base/views/building.xml:16]: 👍

File: housing_cooperative_base/models/lease.py [hunk 3]
```diff
@@ -162,26 +162,25 @@ def create_contract(self):
                     'recurring_invoices': True,
                     'lease_id': self.id,
                     'journal_id': self._default_journal().id,
-            })
-        )
+                }))
```
Comment (robinkeunen) [inline housing_cooperative_base/models/lease.py:172]: mmh, matter of taste on that one: did you go through **black**?

File: housing_cooperative_base/models/room.py [hunk 4]
```diff
@@ -50,3 +55,39 @@ def _compute_state(self):
                 room.state = 'busy'
             else:
                 room.state = 'available'
+
+    @api.onchange('cluster_id')
+    def onchange_cluster_id(self):
+        if self.housing_id and self.cluster_id:
+            self.housing_id = None
+            return {
+                'warning': {
```
Comment (robinkeunen) [inline housing_cooperative_base/models/room.py]: Nice, c'est nouveau cette façon de renvoyer des warnings. Tu m'apprends un truc.

File: housing_cooperative_base/views/housing.xml [hunk 5]
```diff
@@ -36,21 +36,14 @@
...
                             <group>
-                                <field name="lease_ids"  nolabel="1">
-                                    <tree>
-                                        <field name="tenant_id"/>
-                                        <field name="start"/>
-                                        <field name="end"/>
-                                        <field name="state"/>
-                                    </tree>
-                                </field>
+                                <field name="room_ids" options="{'no_create': True}" widget="many2many"/>
```
Comment (robinkeunen) [inline housing_cooperative_base/views/housing.xml]: If you don't set the columns, odoo will use the tree view with the highest priority, here `lease_view_tree`: 

```xml 
     <record id="lease_view_tree" model="ir.ui.view">
         <field name="name">lease_view_tree</field>
         <field name="model">hc.lease</field>
         <field name="arch" type="xml">
             <tree string="lease_tree">
                 <field name="housing_id"/>
                 <field name="start"/>
                 <field name="end"/>
                 <field name="tenant_id"/>
                 <field name="state"/>
             </tree>
         </field>
     </record>
```

Which has a useless `housing_id` column (we're on the housing form) and has an order less useful here (`tenant_id` first seems more logical to me)

File: housing_cooperative_base/models/res_partner.py [hunk 6]
```diff
@@ -32,6 +32,10 @@ class ResPartner(models.Model):
         comodel_name='res.country',
         string='Citizenship',
         ondelete='restrict')
+    study = fields.Many2one(
+        comodel_name='hc.studies',
+        string='Study',
+        ondelete='restrict')
```
Comment (robinkeunen) [inline housing_cooperative_base/models/res_partner.py:42]: Never took the time to understand, maybe you can clarify  the three options (set null, cascade, restrict) for me:

- Which is default?
- What happens if I delete a study object?
- What happens if i delete a partner object?

**Review by robinkeunen — COMMENTED** [review]
@mclaeysb Good job, it's clean and mostly correct: check my comments.

More generally about contributing:

- make PR from your branches to _housing-coop-base_, it will make reviews easy for me and I can approve them as you go. A PR by task\* is usually a good rule of thumb.
  - \*task in the Coop IT Easy kanban
- be more explicit on the commit messages, for example:
  - `[FIX] make better view` could be `[IMP] hcb: add columns to housing tree view`
    - IMP for improvement
    - hcb for the module initials
  - we try to follow [OCA guidelines](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#commit-message). I don't respect commit message length recommendation though. I make the message short but 50 characters is way too short.

File: housing_cooperative_base/views/cluster.xml (hunk 1 above)
Comment (mclaeysb) [reply housing_cooperative_base/views/cluster.xml]: C'est un One2many par défault, comme la relation est de ce type!

File: housing_cooperative_base/models/lease.py (hunk 3 above)
Comment (mclaeysb) [reply housing_cooperative_base/models/lease.py:172]: Didn't configure `black` yet. This probably happened unintentionally along the way.

File: housing_cooperative_base/views/housing.xml (hunk 5 above)
Comment (mclaeysb) [reply housing_cooperative_base/views/housing.xml]: I know! There were two leases pages so I changed one to rooms (the other can be found below but was invisible is this shorter diff). Hence my changes..

**Comment by robinkeunen** [comment]
Tu commences à pas mal accumuler les `fixup`, c'est cool mais n'hésite pas à faire  un `rebase -i --autosquash` de temps en temps. Sinon, tu risques de te tirer les cheveux sur les conflits.

**Pull request merged**